### PR TITLE
Rewrite series graph legends in HTML instead of SVG

### LINKED ISSUE #3019: Reimplement series graph legends in html
The bottom legends in the graphs from #2598 are implemented using svg. This causes problems with text reflow on narrow screens. It would be better to move these legends to a normal div that's positioned in the same place as the current legend.

### PR DESCRIPTION
**Pull request by tibdhond**
The visualisation legends have been rewritten into html (instead of svg). Several styling attributes have been moved to the stylesheet. I also used the opportunity to give the colour boxes rounded colours since I feel like it looks a bit better.

image

image

Closes #3019 .

### CODE REVIEW

**Review by chvp — COMMENTED** [review]
Can we not generate the legends in the template files now? Or would that make coloring the indicators too complex?

**Review by bmesuere — CHANGES_REQUESTED** [review]
Text reflow with many items/narrow screens (one of the main reasons of this PR) isn't fixed.

image

The squares should always be square, only the labels should shrink.

**Comment by tibdhond** [comment]
@chvp For the stacked status graph this is probably possible, but it'll probably be a lot harder to for the cumulative graph since the amount of legend items can vary.  I'm not sure if there's an easy/elegant fix for this using ruby/css.

**Comment by bmesuere** [comment]
> @chvp For the stacked status graph this is probably possible, but it'll probably be a lot harder to for the cumulative graph since the amount of legend items can vary. I'm not sure if there's an easy/elegant fix for this using ruby/css.

I'm ok with doing this in JS

**Comment by tibdhond** [comment]
A 'flex-wrap' seems to have fixed both the text and the square issue, but I also added a 'min-width' to the boxes for good measure.

File: app/assets/stylesheets/components/visualisations.css.scss [hunk 1]
```diff
@@ -35,3 +36,29 @@
   fill: none;
   stroke-width: 2;
 }
+
+.legend {
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  flex-wrap: wrap;
```
Comment (bmesuere) [inline app/assets/stylesheets/components/visualisations.css.scss:44]:
```suggestion
  flex-wrap: wrap;
  overflow: hidden;
```

File: app/assets/stylesheets/components/visualisations.css.scss [hunk 2]
```diff
@@ -35,3 +36,29 @@
...
+
+  .legend-item {
+    margin-left: 8px;
+    margin-right: 8px;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+
+    .legend-box {
+      min-width: 15px;
```
Comment (bmesuere) [inline app/assets/stylesheets/components/visualisations.css.scss:54]:
```suggestion
```

**Review by bmesuere — APPROVED** [review]

**Pull request merged**